### PR TITLE
Feature#2601 Check RoutingResult.

### PR DESCRIPTION
**Pull request by sunbufu**
Fixes https://github.com/apache/incubator-shardingsphere/issues/2601.

Changes proposed in this pull request:
1. add checker in ParsingSQLRouter for check RoutingResult.
2. add ParsingSQLRoutingResultChecker to check RoutingResult with ShardingRule.RoutingUnits.
3. add, modify some UTs.

### CODE REVIEW

**Comment by sunbufu** [comment]
Reflect is only in AbstractSpringJUnitTest.java for unit test to reload MetaData after create table.

**Comment by sunbufu** [comment]
We can discuss which sql should be check. In my view, we can divide the check into two parts, dataSource check and table check. In this PR table check for DML, and dataSource check for no master-slaver.

**Comment by tuohai666** [comment]
It's right to check all the DMLs. You can take a performance test for comparision.

**Comment by sunbufu** [comment]
I run jmeter with `SELECT * FROM t_order WHERE order_id = 3` (2 threads, 100 loop counts) and find there was not performance impact basically. Because there is no `String + String` operation when check pass, and JVM will optimize `String + String` operation.

**Comment by tuohai666** [comment]
> I run jmeter with `SELECT * FROM t_order WHERE order_id = 3` (2 threads, 100 loop counts) and find there was not performance impact basically. Because there is no `String + String` operation when check pass, and JVM will optimize `String + String` operation.

OK.
Please resolve the coverage decreasing.

File: sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/sharding/ParsingSQLRoutingResultChecker.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+            return false;
+        }
+        if (checkTable && (null == metaData.getTables() || !metaData.getTables().containsTable(actualTableName))) {
+            return false;
+        }
+        return true;
+    }
+    
+    private void throwExceptionForAbsentDataNode(final Collection<String> absentDataNodes, final CharSequence detail) {
+        String msg = "We get some absent DataNodes=" + absentDataNodes + " in routing result, "
```
Comment (tuohai666) [inline sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/sharding/ParsingSQLRoutingResultChecker.java]: Please don't start a new line until the current line reaches 200 characters.

File: sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/sharding/ParsingSQLRoutingResultChecker.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        if (null != shardingRule.getRuleConfiguration().getDefaultDatabaseShardingStrategyConfig()) {
+            databaseStrategy = shardingRule.getRuleConfiguration().getDefaultDatabaseShardingStrategyConfig();
+        }
+        if (null != shardingRule.getRuleConfiguration().getDefaultTableShardingStrategyConfig()) {
+            tableStrategy = shardingRule.getRuleConfiguration().getDefaultTableShardingStrategyConfig();
+        }
+        for (TableUnit each : absentRoutingUnitMap.get(routingUnit)) {
+            absentDataNodes.add(routingUnit.getDataSourceName() + "." + each.getActualTableName());
+            Optional<TableRuleConfiguration> tableRuleConfiguration = getTableRuleConfiguration(shardingRule.getRuleConfiguration().getTableRuleConfigs(), each.getLogicTableName());
+            if (!tableRuleConfiguration.isPresent()) {
```
Comment (tuohai666) [inline sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/sharding/ParsingSQLRoutingResultChecker.java]: Better this way:

            if (tableRuleConfiguration.isPresent()) {
                databaseStrategy = tableRuleConfiguration.get().getDatabaseShardingStrategyConfig();
                tableStrategy = tableRuleConfiguration.get().getTableShardingStrategyConfig();
            }

File: sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/sharding/ParsingSQLRoutingResultChecker.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            tableStrategy = tableRuleConfiguration.get().getTableShardingStrategyConfig();
+        }
+        if (!absentDataNodes.isEmpty()) {
+            StringBuilder detail = new StringBuilder();
+            if (null != databaseStrategy) {
+                detail.append("DatabaseStrategy=[").append(databaseStrategy).append("], ");
+            }
+            detail.append("TableStrategy=[").append(tableStrategy).append("], ");
+            if (shardingStatement instanceof ShardingConditionOptimizedStatement) {
+                detail.append("with ")
```
Comment (tuohai666) [inline sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/sharding/ParsingSQLRoutingResultChecker.java]: Please don't start a new line until the current line reaches 200 characters.

**Pull request merged**